test/lib/bamboo/adapters/sparkpost_adapter_test.exs

```diff
@@ -0,0 +1,135 @@
+defmodule Bamboo.SparkpostAdapterTest do
+  use ExUnit.Case
+  alias Bamboo.Email
+  alias Bamboo.SparkpostAdapter
+
+  @config %{adapter: SparkpostAdapter, api_key: "123_abc"}
+  @config_with_bad_key %{adapter: SparkpostAdapter, api_key: nil}
+
+  defmodule FakeSparkpost do
+    use Plug.Router
+
+    plug Plug.Parsers,
+      parsers: [:urlencoded, :multipart, :json],
+      pass: ["*/*"],
+      json_decoder: Poison
+    plug :match
+    plug :dispatch
+
+    def start_server(parent) do
+      Agent.start_link(fn -> HashDict.new end, name: __MODULE__)
+      Agent.update(__MODULE__, &HashDict.put(&1, :parent, parent))
+      Application.put_env(:bamboo, :mandrill_base_uri, "http://localhost:4001")
+      Plug.Adapters.Cowboy.http __MODULE__, [], port: 4001, ref: __MODULE__
```

Comment:
You may want to use the changes from this commit: https://github.com/thoughtbot/bamboo/commit/1de0bc49e624beef9a5607d139aafeb99489615e 

Previously there were intermittent failures because a port wouldn't fully close before a new test started. The changes in that commit ensure a free port is always used :)